instrumentation/reactor-netty/reactor-netty-1.0/javaagent/src/test/groovy/AbstractReactorNettyHttpClientTest.groovy

```diff
@@ -52,4 +61,87 @@ abstract class AbstractReactorNettyHttpClientTest extends HttpClientTest<HttpCli
   }
 
   abstract HttpClient createHttpClient()
+
+  def "should expose context to http client callbacks"() {
+    given:
+    def onRequestSpan = new AtomicReference<Span>()
+    def afterRequestSpan = new AtomicReference<Span>()
+    def onResponseSpan = new AtomicReference<Span>()
+    def afterResponseSpan = new AtomicReference<Span>()
+
+    def httpClient = createHttpClient()
+      .doOnRequest({ rq, con -> onRequestSpan.set(Span.current()) })
+      .doAfterRequest({ rq, con -> afterRequestSpan.set(Span.current()) })
+      .doOnResponse({ rs, con -> onResponseSpan.set(Span.current()) })
+      .doAfterResponseSuccess({ rs, con -> afterResponseSpan.set(Span.current()) })
+
+    when:
+    runUnderTrace("parent") {
+      httpClient.baseUrl(server.address.toString())
+        .get()
+        .uri("/success")
+        .response()
+        .block()
+    }
+
+    then:
+    assertTraces(1) {
+      trace(0, 3) {
+        def parentSpan = span(0)
+        def nettyClientSpan = span(1)
+
+        basicSpan(it, 0, "parent")
+        clientSpan(it, 1, parentSpan, "GET", server.address.resolve("/success"))
+        serverSpan(it, 2, nettyClientSpan)
+
+        assertSameSpan(parentSpan, onRequestSpan)
+        assertSameSpan(nettyClientSpan, afterRequestSpan)
+        assertSameSpan(nettyClientSpan, onResponseSpan)
+        assertSameSpan(parentSpan, afterResponseSpan)
+      }
+    }
+  }
+
+  def "should expose context to http request error callback"() {
```

Comment:
can this test be added to 0.9? (main reason i'm interested is to see which span it uses, client or parent)

Reply:
Hmm, weird - I remember that I had some problems with that test on 0.9, I remember being convinced that this test made absolutely no sense in 0.9... but now that I copied it from 1.0 it just passed without any changes 😅